src/Tasks/SystemState.cs

```diff
@@ -596,6 +606,73 @@ out fileState.frameworkName
             frameworkName = fileState.frameworkName;
         }
 
+        /// <summary>
+        /// Reads in cached data from stateFiles to build an initial cache. Avoids logging warnings or errors.
+        /// </summary>
+        /// <param name="stateFiles">List of locations of caches on disk.</param>
+        /// <param name="log">How to log</param>
+        /// <param name="fileExists">Whether a file exists</param>
+        /// <returns>A cache representing key aspects of file states.</returns>
+        internal static SystemState DeserializePrecomputedCachesByTranslator(ITaskItem[] stateFiles, TaskLoggingHelper log, FileExists fileExists)
+        {
+            SystemState retVal = new SystemState();
+            retVal.isDirty = stateFiles.Length > 0;
+            HashSet<string> assembliesFound = new HashSet<string>();
+
+            foreach (ITaskItem stateFile in stateFiles)
+            {
+                // Verify that it's a real stateFile. Log message but do not error if not.
+                SystemState sysState = DeserializeCacheByTranslator(stateFile.ToString(), log, false);
+                if (sysState == null)
+                {
+                    continue;
+                }
+                foreach (KeyValuePair<string, FileState> kvp in sysState.instanceLocalFileStateCache)
+                {
+                    string relativePath = kvp.Key;
+                    if (!assembliesFound.Contains(relativePath))
```

Comment:
Worth logging a warning if this assembly has been seen before? Or is the expectation that there will be conflicts and the user needs to put the more important files first?

Reply:
There may be conflicts if, say, two precomputed caches are created (and used) that both include My.Assembly.dll.

It shouldn't be a problem, however, because there should only be one assembly at that relative path, so it should have identical data from each. The original version of this verified that via an mvid check; I since simplified it, so theoretically someone could put incorrect information into their cache and prevent correct information from being read in, but that's just an error with the first cache rather than anything about ordering, and it doesn't break the build or anything, just making your build slightly longer. Given the above, I'd prefer to leave it as-is.